### PR TITLE
#45 added save functionality to scan

### PR DESCRIPTION
**Pull request by ammaratef45**
PR for #45

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#92](https://github.com/ammaratef45/attendance/issues/92) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#92](https://github.com/ammaratef45/attendance/issues/92) is assigned to @bahaaib[/z](https://www.0crat.com/u/bahaaib), here is [why](https://www.0crat.com/footprint/CH951175M/c223d8a1-5b28-4c47-8921-1e7e8028a582); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @ammaratef45[/z](https://www.0crat.com/u/ammaratef45) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; @bahaaib[/z](https://www.0crat.com/u/bahaaib) is not a member of this project yet, but they can request to join, as [§1](http://www.zerocracy.com/policy.html#1) explains; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

File: attendance/lib/backend/scan.dart [hunk 1]
```diff
@@ -68,6 +74,33 @@ class Scan {
...
+        await User.token()
+      );
+      _markSaved();
+    } on Exception catch (e) {
+      print(e.toString());
+    }
+  }
+
+  void _persist() {
+    DBProvider.db.newScan(this);
```
Comment (Bahaaib) [inline attendance/lib/backend/scan.dart]: @ammaratef45 Don't you think it'd be more expressive and if 'newScan' renamed to 'initScan'? 
It'd be less confusing with the original 'new' keyword in readability, i believe!

Comment (ammaratef45) [reply attendance/lib/backend/scan.dart]: @Bahaaib I agree with you, yet I will edit it and add more unit tests.
Also, The puzzle isn't deleted, I will wait to merge #90 first :)

Comment (Bahaaib) [reply attendance/lib/backend/scan.dart]: @ammaratef45 OKay

**Comment by ammaratef45** [comment]
@Bahaaib Can you have a look now? Thanks.

**Comment by Bahaaib** [comment]
@rultor merge

**Comment by ammaratef45** [comment]
@rultor merge

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31):

**Comment by 0crat** [comment]
Job was finished in 28 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#92](https://github.com/ammaratef45/attendance/issues/92) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

**Pull request merged**